pkg/controller/agent/vlanconfig/controller.go

```diff
@@ -178,31 +179,26 @@ func matchClusterNetwork(vc *networkv1.VlanConfig, vs *networkv1.VlanStatus) boo
 	return vc.Spec.ClusterNetwork == vs.Status.ClusterNetwork
 }
 
+// only sets up uplink & vlan bridge, vids are added by clusternetwork controller
 func (h Handler) setupVLAN(vc *networkv1.VlanConfig) error {
 	var v *vlan.Vlan
 	var setupErr error
-	var localAreas []*vlan.LocalArea
 	var uplink *iface.Link
-	// get VIDs
-	localAreas, setupErr = h.getLocalAreas(vc)
-	if setupErr != nil {
-		goto updateStatus
-	}
+
 	// construct uplink
 	uplink, setupErr = setUplink(vc)
 	if setupErr != nil {
 		goto updateStatus
 	}
-	// set up VLAN
+	// set up VLAN bridge
 	v = vlan.NewVlan(vc.Spec.ClusterNetwork)
 	if setupErr = v.Setup(uplink); setupErr != nil {
 		goto updateStatus
 	}
-	setupErr = h.AddLocalAreas(v, localAreas)
 
```

Comment:
earlier if had error setting up bridge interface we would skip the setup of local network using `goto updateStatus`

now that this is not needed we could just avoid the entire usage of jump/label in the flow